src/overlays/actors/ovl_Bg_Lotus/z_bg_lotus.c

```diff
@@ -20,22 +22,154 @@ const ActorInit Bg_Lotus_InitVars = {
     (ActorFunc)BgLotus_Init,
     (ActorFunc)BgLotus_Destroy,
     (ActorFunc)BgLotus_Update,
-    (ActorFunc)BgLotus_Draw
+    (ActorFunc)BgLotus_Draw,
 };
-*/
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Lotus/BgLotus_Init.s")
+s32 D_80AD6D10[] = { 0x48580064 };
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Lotus/BgLotus_Destroy.s")
+void BgLotus_Init(Actor* thisx, PlayState* play) {
+    BgLotus* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Lotus/func_80AD6830.s")
+    s32 pad;
+    s32 sp2C;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Lotus/func_80AD68DC.s")
+    Actor_ProcessInitChain(&this->dyna.actor, (InitChainEntry*)D_80AD6D10);
+    DynaPolyActor_Init((DynaPolyActor*)this, 1);
+    DynaPolyActor_LoadMesh(play, (DynaPolyActor*)this, &gLilyPadCol);
+    this->dyna.actor.floorHeight = BgCheck_EntityRaycastFloor5(&play->colCtx, &this->dyna.actor.floorPoly, &sp2C,
+                                                               &this->dyna.actor, &this->dyna.actor.world.pos);
+    this->unk168 = 0x60;
+    this->dyna.actor.world.rot.y = ((s32)Rand_Next() >> 0x10);
+    this->unk15C = &func_80AD68DC;
```

Comment:
```suggestion
    this->actionFunc = func_80AD68DC;
```
I know this isn't documented, but this one is super easy: almost every actor has a state system where the actor calls a function pointer with extended functionality determining their state from their update function, this is the `action function` as well call it. 

Usually these actors have these pre-defined already because someone ran a script across the whole repo over a year ago, this actor unfortunately had it removed, but it's easy to spot. Most actors should only have one function pointer, the function pointer is run in the Update function. 

You need to add 
```c
typedef void (*BgLotusActionFunc)(struct BgLotus*, PlayState*);
``` 
to your header, then change unk 15c to
```c
/* 0x15C */ BgLotusActionFunc actionFunc;
```
and you should be good to go.